src/Xamarin.Android.Build.Tasks/Utilities/Aapt2Daemon.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks {
+	class Aapt2Daemon : IDisposable {
+		struct DaemonProcess {
+			public Process Process;
+			public object LockObject;
```

Comment:
I would make this:
```suggestion
			public readonly object LockObject = new object ();
```
So you don't have to worry about assigning it or overwriting it.

Reply:
The `DaemonProcess` is a struct, so we can't use that. I could make it a class. However its such a small object I'm not sure its worth it.